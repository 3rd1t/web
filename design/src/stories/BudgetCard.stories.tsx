import React from "react";
import { BudgetCard } from '../components/BudgetCard';

export default {
  title: "Budget Card",
};

export const card = () => {

  return (
    <BudgetCard  expensesCount= {195} totalBudget={1500.00} start={new Date()} end={new Date()} budgeted={4.99} expenses={625.12} income={10.00}/>
  )
};
