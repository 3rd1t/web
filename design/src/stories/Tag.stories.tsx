import React from 'react';
import Tag from "../components/Tag"
export default { title: 'Tag' };

export const Default = () => {
    return (
        <Tag>Hello Tag</Tag>
    )
} 
