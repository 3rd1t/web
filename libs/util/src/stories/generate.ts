/**
 * Return all components from storybook.
 *
 * @example
 * import * as stories from "./<component>.stories"
 *
 * const testCases = generateTestCasesFromStories(stories)
 * testCases.forEach((s) => {
 *   it(s.name + "should match snapshot", () => {
 *     const { baseElement } = render(s.component())
 *     expect(baseElement).toMatchSnapshot()
 *   })
 * })
 *
 *
 * @param stories - import * as stories from "./<component>.stories"
 * @returns Name and render function for each component
 */
export function generateTestCasesFromStories(
  stories: Record<string, any>, // eslint-disable-line
): { name: string; component: () => HTMLElement }[] {
  return Object.entries(stories)
    .filter((s) => s[0] !== "default")
    .map((s) => {
      return {
        name: s[0],
        component: s[1],
      }
    })
}
