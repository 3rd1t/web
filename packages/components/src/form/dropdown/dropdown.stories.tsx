import React from "react"
import { Dropdown, DropdownProps } from "./dropdown"

export default {
  component: Dropdown,
  title: "Dropdown",
}

export const primary = () => {
  /* eslint-disable-next-line */
  const props: DropdownProps = {
    button: "Open",
    options: [<span>Hello</span>],
  }

  return <Dropdown button={props.button} options={props.options} />
}
