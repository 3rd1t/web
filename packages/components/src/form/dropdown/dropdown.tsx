import React from "react"
import { Menu, Transition } from "@headlessui/react"

/* eslint-disable-next-line */
export interface DropdownProps {
  button: string | React.ReactNode
  options: React.ReactNode[]
}

export const Dropdown = ({ button, options }: DropdownProps) => {
  return (
    <Menu>
      {({ open }) => (
        <div className="relative">
          <Menu.Button className="focus:outline-none">{button}</Menu.Button>
          <Transition
            show={open}
            enter="transition duration-100 ease-out"
            enterFrom="transform scale-95 opacity-0"
            enterTo="transform scale-100 opacity-100"
            leave="transition duration-75 ease-out"
            leaveFrom="transform scale-100 opacity-100"
            leaveTo="transform scale-95 opacity-0"
          >
            <div className="absolute w-full mt-1 bg-white rounded-md shadow-lg">
              <Menu.Items as="ul" className="max-h-56 focus:outline-none sm:text-sm sm:leading-5 ">
                {options.map((option, i) => {
                  return (
                    <Menu.Item
                      as="li"
                      key={i}
                      className="w-full px-4 py-2 text-gray-700 cursor-default select-none hover:text-gray-900 hover:bg-gray-100"
                    >
                      {option}
                    </Menu.Item>
                  )
                })}
              </Menu.Items>
            </div>
          </Transition>
        </div>
      )}
    </Menu>
  )
}

export default Dropdown
