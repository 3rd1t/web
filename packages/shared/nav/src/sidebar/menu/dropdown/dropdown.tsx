import React, { useState } from "react"
import { Title } from "../title/title"
import { motion, AnimatePresence } from "framer-motion"
export interface DropdownProps {
  label: string
  items: React.ReactNode[]
  row?: boolean
}

export const Dropdown = ({ label, items, row }: DropdownProps) => {
  const [isOpen, setOpen] = useState(true)

  return (
    <div>
      <button onClick={() => setOpen(!isOpen)} className="flex items-center justify-between w-full focus:outline-none">
        <Title label={label} />
        <motion.svg
          animate={isOpen ? "open" : "collapsed"}
          variants={{
            open: { rotate: -90 },
            collapsed: { rotate: 0 },
          }}
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          className="w-5 h-5"
        >
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M15 19l-7-7 7-7" />
        </motion.svg>
      </button>
      <AnimatePresence>
        {isOpen ? (
          <motion.div
            className="px-4"
            initial="collapsed"
            animate="open"
            exit="collapsed"
            variants={{
              open: { y: 0, opacity: 1, height: "auto", transition: { staggerChildren: 0.05 } },
              collapsed: { y: -10, opacity: 0, height: 0 },
            }}
          >
            <ul className={row ? "flex justify-between items-center" : ""}>
              {items.map((item, index) => {
                return (
                  <motion.li
                    className="flex justify-between"
                    key={index}
                    variants={{ open: { y: 0, opacity: 100 }, collapsed: { y: -10, opacity: 0 } }}
                  >
                    {item}
                  </motion.li>
                )
              })}
            </ul>
          </motion.div>
        ) : null}
      </AnimatePresence>
    </div>
  )
}

export default Dropdown
