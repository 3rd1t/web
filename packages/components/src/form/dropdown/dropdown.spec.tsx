import { render } from "@testing-library/react"
import * as stories from "./dropdown.stories"
import { generateTestCasesFromStories } from "@perfolio/util/stories/generate"

describe("Dropdown", () => {
  describe("Stories", () => {
    const testCases = generateTestCasesFromStories(stories)
    testCases.forEach((s) => {
      it(s.name + " should match snapshot", () => {
        const { baseElement } = render(s.component())
        expect(baseElement).toMatchSnapshot()
      })
    })
  })
})
