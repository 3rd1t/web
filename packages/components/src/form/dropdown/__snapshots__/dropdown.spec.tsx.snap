// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Stories primary should match snapshot 1`] = `
<body>
  <div>
    <div
      class="relative"
    >
      <button
        aria-haspopup="true"
        class="focus:outline-none"
        id="headlessui-menu-button-1"
        type="button"
      >
        Open
      </button>
    </div>
  </div>
</body>
`;
